app/src/main/java/org/p2p/wallet/striga/signup/repository/StrigaSignupDataDatabaseRepository.kt

```diff
@@ -70,6 +70,17 @@ class StrigaSignupDataDatabaseRepository(
         )
     }
 
+    override suspend fun updateSignupData(newData: Collection<StrigaSignupData>): StrigaDataLayerResult<Unit> = try {
+        val entities = newData.map { mapper.toEntity(it, currentUserPublicKey) }
+        dao.updateOrInsertDataWithTransaction(entities)
+        success()
+    } catch (error: Throwable) {
+        StrigaDataLayerError.from(
+            error = error,
+            default = StrigaDataLayerError.InternalError(error)
+        )
```

Comment:
what error can by thrown here?

Reply:
I assume, in rare cases it could throw something like IO-related errors